pkg/deploy/kubernetes_image_puller.go

```diff
@@ -501,7 +540,41 @@ func GetExpectedKubernetesImagePuller(ctx *DeployContext) *chev1alpha1.Kubernete
 	}
 }
 
-// Unisntall the CSV, OperatorGroup, Subscription, KubernetesImagePuller, and update the CheCluster to remove
+// UpdateDefaultImagesIfNeeded updates the default images from `spec.images` if needed
+func UpdateDefaultImagesIfNeeded(ctx *DeployContext) error {
+	specImages := StringToImageSlice(ctx.CheCluster.Spec.ImagePuller.Spec.Images)
+	defaultImages := GetDefaultImages()
+	if UpdateSpecImages(specImages, defaultImages) {
+		defaultImageString := ImageSliceToString(specImages)
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images = defaultImageString
```

Comment:
Pls find better names here as well.
Other than that pr sounds good.

Reply:
I think it could be just 'images'

Reply:
I created a new function (called twice in the code) for this, with new names. How's this?
https://github.com/eclipse-che/che-operator/blob/3db42c1f1e5149a9bfbaf26e88d065c5220caf49/pkg/deploy/kubernetes_image_puller.go#L574-L579